Built-in inline media controls on macOS do not match AVKit

Phinny01 opened the pull request:
#### c1faece3098b90bf6f860f71fe3e9509a04175e8
<pre>
Built-in inline media controls on macOS do not match AVKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=298497">https://bugs.webkit.org/show_bug.cgi?id=298497</a>
<a href="https://rdar.apple.com/156730307">rdar://156730307</a>

Reviewed by Andy Estes.

Change volume-slider-container size and rounded corners to match AVKit. Move the mute button from
the bottom bar to the volume-slider-container. Create a pill shape thumb to match that of AVKit.
Use Display: flex and postion: relative for media-controls-container and controls-bar.bottom for
mac inline for easier positioning. Add new layout tests for the new media controls specs. Add a
feature flag to hide the changes until continuous corners are available in CSS to use instead of
border radius.

media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container.html
media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled.html
media/track/video-track-add-visible-new.html
* LayoutTests/http/tests/contentextensions/text-track-blocked-expected.txt:
* LayoutTests/http/tests/contentextensions/text-track-blocked.html:
* LayoutTests/media/modern-media-controls/audio/audio-controls-buttons-expected.txt:
* LayoutTests/media/modern-media-controls/audio/audio-controls-buttons.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-layout-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-layout.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container-expected.txt: Copied from LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar-expected.txt.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container.html: Copied from LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar.html.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-placard-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-placard.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-top-right-controls-bar-hidden-when-mute-button-disabled-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-top-right-controls-bar-hidden-when-mute-button-disabled.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled-expected.txt: Added.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled.html: Added.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-visibility-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-visibility.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-styles-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-styles.html:
* LayoutTests/media/modern-media-controls/media-controller/media-controller-click-on-video-controls-should-not-pause.html:
* LayoutTests/media/modern-media-controls/volume-support/volume-support-drag-to-mute.html:
* LayoutTests/media/track/video-track-add-visible-new-expected.txt: Added.
* LayoutTests/media/track/video-track-add-visible-new.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::isMediaControlsMacInlineSizeSpecsEnabled const):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.idl:
* Source/WebCore/Modules/modern-media-controls/controls/macos-inline-media-controls.css:
(.media-controls.mac.inline.audio .volume-slider-container):
(.media-controls.mac.inline .volume-slider-container &gt; .background-tint): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/macos-inline-media-controls.js:
(MacOSInlineMediaControls.prototype._setupVolumeSliderConfiguration):
(MacOSInlineMediaControls.prototype.findHostFromParent):
(MacOSInlineMediaControls.prototype.get mediaControlsHost):
(MacOSInlineMediaControls.prototype.layout):
(MacOSInlineMediaControls.prototype.handleEvent):
(MacOSInlineMediaControls.prototype.rightContainerButtons):
(MacOSInlineMediaControls):
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.css:
(:host(audio) .media-controls.mac,): Deleted.
(.media-controls.mac.MacInlineSizeSpecs .controls-bar.bottom,): Deleted.
(.media-controls.mac.MacInlineSizeSpecs .controls-bar.bottom .background-tint,): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/slider.js:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::macOSInlineMediaControlsStyleSheet):
(WebCore::RenderThemeCocoa::mediaControlsStyleSheets):

Canonical link: <a href="https://commits.webkit.org/301493@main">https://commits.webkit.org/301493@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f388dd7c83daaa0cf9f5ac1fcd733f77affa8aec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45166 "Built successfully") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77450 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0cbac85e-49e5-4295-b307-6c172a642615) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127374 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45852 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53727 "Built successfully") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63528 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/42aacea9-4a3d-4532-bd70-a9e39e1a111a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36645 "Passed tests") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/49c176b0-ee00-4b2d-adc8-12a10f5864e6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35541 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30410 "Passed tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106418 "Passed tests") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/124015 "Built successfully and passed tests") | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-2-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40074 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104051 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52745 "Built successfully") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26570 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49160 "Passed tests") | [⏳ 🧪 mac-intel-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 playstation ](https://ews-build.webkit.org/#/builders/PlayStation-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-18-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57991 "Built successfully") | [❌ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157030 "") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51548 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/157030 "") | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-11-Build-EWS "Waiting in queue, processing has not started yet") | | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-11-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/048253bdf9764045ecbca6d696e7248618b9daaa. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50410

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/50cd00d4b419b22485c11d73cbe31acfff2e6209. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50410

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a47d034aa441571d220663906f604b44bf2f7e90. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50410

Commit:
Built-in inline media controls on macOS do not match AVKit
https://bugs.webkit.org/show_bug.cgi?id=298497
rdar://156730307

Reviewed by Andy Estes.

Change volume-slider-container size and rounded corners to match AVKit. Move the mute button from
the bottom bar to the volume-slider-container. Create a pill shape thumb to match that of AVKit.
Use Display: flex and postion: relative for media-controls-container and controls-bar.bottom for
mac inline for easier positioning. Add new layout tests for the new media controls specs. Add a
feature flag to hide the changes until continuous corners are available in CSS to use instead of
border radius.

media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container.html
media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled.html
media/track/video-track-add-visible-new.html
* LayoutTests/http/tests/contentextensions/text-track-blocked-expected.txt:
* LayoutTests/http/tests/contentextensions/text-track-blocked.html:
* LayoutTests/media/modern-media-controls/audio/audio-controls-buttons-expected.txt:
* LayoutTests/media/modern-media-controls/audio/audio-controls-buttons.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-layout-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-layout.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container-expected.txt: Copied from LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar-expected.txt.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-volume-slider-container.html: Copied from LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-mute-button-in-bottom-or-top-right-controls-bar.html.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-placard-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-placard.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-top-right-controls-bar-hidden-when-mute-button-disabled-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-top-right-controls-bar-hidden-when-mute-button-disabled.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled-expected.txt: Added.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-container-hidden-when-mute-button-disabled.html: Added.
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-visibility-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-slider-visibility.html:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-styles-expected.txt:
* LayoutTests/media/modern-media-controls/macos-inline-media-controls/macos-inline-media-controls-volume-styles.html:
* LayoutTests/media/modern-media-controls/media-controller/media-controller-click-on-video-controls-should-not-pause.html:
* LayoutTests/media/modern-media-controls/volume-support/volume-support-drag-to-mute.html:
* LayoutTests/media/track/video-track-add-visible-new-expected.txt: Added.
* LayoutTests/media/track/video-track-add-visible-new.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::isMediaControlsMacInlineSizeSpecsEnabled const):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.idl:
* Source/WebCore/Modules/modern-media-controls/controls/macos-inline-media-controls.css:
(.media-controls.mac.inline.audio .volume-slider-container):
(.media-controls.mac.inline .volume-slider-container > .background-tint): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/macos-inline-media-controls.js:
(MacOSInlineMediaControls.prototype._setupVolumeSliderConfiguration):
(MacOSInlineMediaControls.prototype.findHostFromParent):
(MacOSInlineMediaControls.prototype.get mediaControlsHost):
(MacOSInlineMediaControls.prototype.layout):
(MacOSInlineMediaControls.prototype.handleEvent):
(MacOSInlineMediaControls.prototype.rightContainerButtons):
(MacOSInlineMediaControls):
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.css:
(:host(audio) .media-controls.mac,): Deleted.
(.media-controls.mac.MacInlineSizeSpecs .controls-bar.bottom,): Deleted.
(.media-controls.mac.MacInlineSizeSpecs .controls-bar.bottom .background-tint,): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/slider.js:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::macOSInlineMediaControlsStyleSheet):
(WebCore::RenderThemeCocoa::mediaControlsStyleSheets):

Canonical link: https://commits.webkit.org/301493@main
